transport/src/main/java/io/netty/channel/DefaultChannelHandlerInvoker.java

```diff
@@ -506,13 +419,18 @@ private static WriteTask newInstance(
             task.ctx = ctx;
             task.msg = msg;
             task.promise = promise;
-            task.size = ((AbstractChannel) ctx.channel()).estimatorHandle().size(msg)
-                    + estimateSize(task, CLASS_SIZES.get());
-            ChannelOutboundBuffer buffer = ctx.channel().unsafe().outboundBuffer();
-            // Check for null as it may be set to null if the channel is closed already
-            if (buffer != null) {
-                buffer.incrementPendingOutboundBytes(task.size);
+            task.size = 0;
```

Comment:
@normanmaurer Fixed, although I don't think that the size of a task written to a closed channel matters much.